src/main/java/de/tum/in/www1/artemis/web/rest/FileUploadSubmissionResource.java

```diff
@@ -311,16 +292,11 @@ public FileUploadSubmissionResource(SubmissionRepository submissionRepository, R
         }
 
         Optional<Submission> optionalSubmission = participation.findLatestSubmission();
-        FileUploadSubmission fileUploadSubmission;
-        if (optionalSubmission.isEmpty()) {
-            // this should never happen as the submission is initialized along with the participation when the exercise is started
-            fileUploadSubmission = new FileUploadSubmission();
-            fileUploadSubmission.setParticipation(participation);
-        }
-        else {
-            // only try to get and set the file upload if the fileUploadSubmission existed before
-            fileUploadSubmission = (FileUploadSubmission) optionalSubmission.get();
-        }
+        FileUploadSubmission fileUploadSubmission = (FileUploadSubmission) optionalSubmission.orElseGet(() -> {
+            FileUploadSubmission tempParticipation = new FileUploadSubmission();
+            tempParticipation.setParticipation(participation);
+            return tempParticipation;
+        });
```

Comment:
For the next lines: i do not understand how we ensure, that only the latest submission and result are sent by setting the participation submission and result to null?